libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -425,21 +440,35 @@ static void expectProcessLoopCalls( MQTTContext_t * const pContext,
     {
         if( incomingPublish )
         {
-            MQTT_UpdateStatePublish_ExpectAnyArgsAndReturn( ( stateAfterDeserialize == MQTTStateNull )?
-                                                            MQTTIllegalState : MQTTSuccess );
+            MQTT_UpdateStatePublish_ExpectAnyArgsAndReturn( updateStateStatus );
             MQTT_UpdateStatePublish_ReturnThruPtr_pNewState( &stateAfterDeserialize );
         }
         else
         {
-            MQTT_UpdateStateAck_ExpectAnyArgsAndReturn( ( stateAfterDeserialize == MQTTStateNull )?
-                                                        MQTTIllegalState : MQTTSuccess );
+            MQTT_UpdateStateAck_ExpectAnyArgsAndReturn( updateStateStatus );
             MQTT_UpdateStateAck_ReturnThruPtr_pNewState( &stateAfterDeserialize );
         }
 
         if( stateAfterDeserialize == MQTTPublishDone )
         {
             expectMoreCalls = false;
         }
+        else
+        {
+            switch( updateStateStatus )
+            {
+                case MQTTSuccess:
+                    expectMoreCalls = true;
+                    break;
+
+                case MQTTStateCollision:
+                    expectMoreCalls = ( pPubInfo->dup == true ) ? true : false;
```

Comment:
We don't need to worry about the extra MISRA code for this file
```suggestion
                    expectMoreCalls = pPubInfo->dup;
```